src/db/models/user.model.js

```diff
@@ -172,5 +264,40 @@ module.exports = (sequelize, DataTypes) => {
             });
     };
 
+    User.prototype.updateVerifyAccountToken = function updateVerifyAccountToken(email, expTime) {
+        return randomBytes(20)
+            .then((buf) => {
+                const token = buf.toString('hex');
+                return token;
+            }).then((token) => {
+                this.contactMethodVerificationToken = token;
+                this.contactMethodToVerify = email;
+                const m = moment.utc();
+                m.add(expTime, 'seconds');
+                this.contactMethodVerificationTokenExpires = m.format();
+                return this.save().then(() => token);
+            });
+    };
+
+    User.prototype.addVerifiedContact = function addVerifiedContact(contactMethodToVerify) {
+        if (this.verifiedContactMethods.includes(contactMethodToVerify)) {
+            // Handle instances where the user contact already exsits in the list of verified
+            // contact methods
+            this.contactMethodVerificationToken = null;
+            this.contactMethodVerificationTokenExpires = null;
+            this.contactMethodToVerify = null;
+            return this.save();
+        }
+        const authorizedUsers = this.verifiedContactMethods;
+        const authorizedUsersLength = authorizedUsers.push(contactMethodToVerify.toString());
+        if (authorizedUsersLength > 0) {
```

Comment:
This function will always be called with a `contactMethodToVerify` argument, and therefore this conditional will always evaluate to true, correct?

If this function should always be called with a `contactMethodToVerify` and you want to ensure this is the case, rather than putting this conditional here, put one that the top of the function and throw or return an error if this function is called without a `contactMethodToVerify`

This whole thing seems overly verbose.

Reply:
yea, can probably be removed